featurebyte/models/use_case.py

```diff
@@ -0,0 +1,60 @@
+"""
+Use Case model
+"""
+from typing import List, Optional
+
+import pymongo
+from pydantic import Field
+
+from featurebyte.models.base import (
+    FeatureByteCatalogBaseDocumentModel,
+    PydanticObjectId,
+    UniqueConstraintResolutionSignature,
+    UniqueValuesConstraint,
+)
+
+
+class UseCaseModel(FeatureByteCatalogBaseDocumentModel):
+    """
+    UseCaseModel represents a use case within a feature store
+
+    context_id: PydanticObjectId
+        The context id of ContextModel
+    target_id: PydanticObjectId
+        The target id of TargetModel
+    observation_table_ids:  List[PydanticObjectId]
+        The observation table ids of ObservationTableModel
+    default_preview_table_id: PydanticObjectId
+        The default preview table from observation_table_ids
+    default_eda_table_id: PydanticObjectId
+        The default eda table id of from observation_table_ids
+    """
+
+    context_id: PydanticObjectId
+    target_id: PydanticObjectId
+    observation_table_ids: List[PydanticObjectId] = Field(default=[])
+    default_preview_table_id: Optional[PydanticObjectId]
+    default_eda_table_id: Optional[PydanticObjectId]
```

Comment:
```
  observation_table_ids: List[PydanticObjectId] = Field(default_factory=list)
  default_preview_table_id: Optional[PydanticObjectId] = Field(default=None)
  default_eda_table_id: Optional[PydanticObjectId] = Field(default=None)
```

If `None` should be default for the last 2 attributes, we should add `default=None` in the parameter as `Optional` does not imply `default=None` (https://docs.pydantic.dev/latest/usage/fields/#default-values).